testSettings.py

```diff
@@ -37,9 +35,42 @@ def _post_check(self) -> None:
         self._lock: threading.Lock
         object.__setattr__(self, "_lock", threading.Lock())
 
-        # Check that the cfg_descr has a connection/test_case_id
+        # Access some properties here. We would get an exception early on, if
+        # there is something wrong.
         self.connection
-        self.test_case_id
+        self.test_case_id.info
+        self.conf_server
+        self.conf_client
+
+    @property
+    def conf_server(self) -> testConfig.ConfServer:
```

Comment:
Where are conf_client and conf_server functions used?

Reply:
at various places. Already previously.

Reply:
I think these should be internal functions and external users would need to use `conf_server_used`

Reply:
At the beginning (on `main` branch), the `conf_server` already exists as an attribute and is already in active use. The branch does several patches, making it a property, and then reduce its use. There are still actual users.

Probably there should be more patches done, that renames things and moves things around further. But as far as this branch goes, it's already (too) large and overall goes in the right direction.

I'd rather have this merged, and improve there further. For example, `ConfServer` has an attribute "persistent", but `ConfClient` doesn't have it. Note how `conf_server_used` returns the base class `ConfBaseClientServer`, which also doesn't have a "persistent" flag. Maybe the "persistent" flag is anyway not implemented correctly, but saying just use "conf_server_used" everywhere needs at least address what to do about this flag (e.g. move it to the parent class). The point is, many patches can be done after this PR.

Also, sure, I could not start out and do a major renaming patch first. But that only causes me extra work to rebase this branch on top of that, and it's not clear why the renaming must come before the changes that are anyway planned. The branch itself uses names that make sense on top of what we currently have, although, maybe there should be some renaming done there.

Reply:
I see please do a rename in a subsequent PR if you have the cycles. Thank you! We should have descriptive and short names, not just short names for the sake of them being short.

Reply:
Regarding the "persistent" flag. It was intended for the server only since iperf3 -s can run indefinitely, but iperf3 -c is only one-shot. I think this flag is in the properly place, now regarding whether the "persistent" flag is still working correctly, I am not too sure after all the recent changes. I am guessing most people test with "persistent" flag being false.

The purpose of the "persistent" flag being there is such that the user can run further tests without managing the server.